### DISCUSSION TITLE
What framework/language will be used?

### DISCUSSION THREAD

**Discussion by ninjamar** [Piggo4676/Foss-AV#8] [General]
What framework/language will be used? If I were to do this project, I would use a django frontend to satisfy the app/gui side of the app. The backend would include an antivirus script (as a python) module to scan files on the computer. Also using django gives us easy database access so we could store details such as which files have viruses etc.

**Comment by imdevarsh** [comment]
I think that the tool should also work in the terminal
Also doesn't Django work in the browser? Shouldn't it be an actual desktop application?

**Comment by ninjamar** [comment]
We can use a python cli and structure this as a python package. Django works in the browser but we can deploy it to localhost. This is easy because we can write the frontend in html and focus mostly on the backend. Using an alternative framework like electron requires lots of overhead. 

If we use python, we can use c extensions to help out

**Comment by Piggo4676** [comment]
yep, python I reckon. easy to use and install.

**Comment by UniqueName12345** [comment]
The owner has decided! In this current stage, python is going to be used for the backend. But what do we do for the front end?

**Reply by jdev082** [reply]
We could use PyQt6 for the front end. Buttons can be connected to functions.

**Reply by Piggo4676** [reply]
Yeah, that could work! :)

**Reply by UniqueName12345** [reply]
> We could use PyQt6 for the front end. Buttons can be connected to functions.

That could work!
